zk/tests/conftest.py

```diff
@@ -71,8 +122,34 @@ def condition():
     compose_file = os.path.join(HERE, 'compose', 'zk.yaml')
     if [3, 5, 0] <= get_version() < [3, 6, 0]:
         compose_file = os.path.join(HERE, 'compose', 'zk35.yaml')
+        if get_ssl():
+            compose_file = os.path.join(HERE, 'compose', 'zk35_ssl.yaml')
     elif get_version() >= [3, 6, 0]:
         compose_file = os.path.join(HERE, 'compose', 'zk36plus.yaml')
-
-    with docker_run(compose_file, conditions=[condition]):
-        yield get_instance
+        if get_ssl():
+            compose_file = os.path.join(HERE, 'compose', 'zk36plus_ssl.yaml')
+
+    private_key = os.path.join(HERE, 'compose', 'private_key.pem')
+    cert = os.path.join(HERE, 'compose', 'cert.pem')
+    ca_cert = os.path.join(HERE, 'compose', 'ca_cert.pem')
+
+    condition_ssl = [
+        CheckDockerLogs(
+            compose_file,
+            'Starting server',
+        )
+    ]
+
+    if get_ssl():
+        condition = condition_ssl
+    else:
+        condition = [condition_non_ssl]
```

Comment:
I am currently using separate conditions for SSL and non-SSL, since SSL Zookeeper took longer to set up but the `condition_non_ssl` function would have failed before it can run correctly for SSL, and checking docker logs wasn't enough for non-SSL since 1 metric would be missing in the Azure CI.